### PR TITLE
EGI PSG

### PR DESCRIPTION
**Pull request by fraimondo**
This is an attempt.

`_read_segment_file` should be cleaned a bit, although I did not went into the specifics.

I still need to get a small recording (10 seconds) using 256 channels and add it to the MNE testing dataset. That will happen tomorrow. With a long recording (60 minutes), I get the ECG properly.

### CODE REVIEW

**Comment by agramfort** [comment]
I have no experience with this format so I cannot do a functional test

@dengemann @deep-introspection @amonapariciog ok with you?

**Comment by deep-introspection** [comment]
@agramfort It seems to work well here 👍 !

**Comment by fraimondo** [comment]
Keep in mind that nosetests need `test_egi_pns.mff` and `test_egi_pns.mat` in the MNE testing dataset, inside the EEG folder [here](https://github.com/mne-tools/mne-testing-data/tree/master/EGI).

I've uploaded the files [here](https://www.dropbox.com/sh/z6shib5s6t6q9h8/AAA-STQvxZYzbJVEbU6cMOnYa?dl=0)

The ZIP contains the MFF, since it is indeed a folder.

**Comment by larsoner** [comment]
@fraimondo can you make a PR to `mne-tools/mne-testing-data`?

**Comment by fraimondo** [comment]
@larsoner PR on the way

**Comment by fraimondo** [comment]
So I just recorded a longer test file that includes several internal blocks, and there is no difference between Matlab and MFF files. I'm not including them in the dataset, but I wanted to be sure that everything is working properly.

I just added the testing dataset 0.40 to this PR.

Functionally, the code is working properly. Now it's on you to review, suggest, comment and if you like how it is written, merge.

File: mne/io/egi/egimff.py [hunk 1]
```diff
@@ -68,10 +70,47 @@ def _read_mff_header(filepath):
...
+                           pns_fname=all_files['PNS']['signal'],
+                           pns_sample_blocks=pns_blocks)
+
+    info_filepath = op.join(filepath, 'info.xml')  # add with filepath
     tags = ['mffVersion', 'recordTime']
     version_and_date = _extract(tags, filepath=info_filepath)
-    summaryinfo.update(version=version_and_date['mffVersion'][0],
+    version = ""
+    if len(version_and_date['mffVersion']):
+        version = version_and_date['mffVersion'][0]
```
Comment (agramfort) [inline mne/io/egi/egimff.py]: it means we can close #4574 right?

**Comment by agramfort** [comment]
code LGTM

+1 for MRG if CIs are green

File: mne/io/egi/egimff.py (hunk 1 above)
Comment (fraimondo) [reply mne/io/egi/egimff.py]: Indeed It adresses the same issues and includes those fixes too.

**Comment by fraimondo** [comment]
Travis is happy, circle fails, as usual.

**Comment by agramfort** [comment]
I have no idea why circle fails

@larsoner ?

no objection to merge on my side

**Comment by larsoner** [comment]
Looks like some `sphinx-gallery` bug. I'll see if I can reproduce it locally.

**Comment by larsoner** [comment]
Found the bug, fix to SG coming but it might take a while to land. In the meantime we can manually check the CircleCI log to make sure this is the error:
```
embedding documentation hyperlinks...

Exception occurred:
  File "/home/ubuntu/mne-python/sphinx-gallery/sphinx_gallery/docs_resolv.py", line 50, in _get_data
    data = StringIO(data)
TypeError: initial_value must be unicode or None, not str
```
And the artifacts are still created I think so we can still inspect outputs.

**Comment by agramfort** [comment]
ok let's merge then

**Comment by larsoner** [comment]
I'll take a look and merge if all is good

**Comment by larsoner** [comment]
Thanks @fraimondo

**Comment by fraimondo** [comment]
Next one is the eximia, as soon as they bring back the equipment to the
hospital.

**Comment by larsoner** [comment]
Feel free to give this PR a try

https://github.com/mne-tools/mne-python/pull/3523

IIRC the only reason we gave up on that PR is a lack of testers / test data

**Pull request merged**